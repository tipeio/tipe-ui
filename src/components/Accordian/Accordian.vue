<template>
  <div class="content">
    <div 
      class="row" 
      @click="onClick">
      <p class="title" >{{ text.toUpperCase() }}</p>
      <tipe-icon
        :class="{open}"
        icon="arrow-down"
        class="icon"
      />
    </div>
    <slot v-if="open"/>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../Icon'

export default {
  name: 'Accordian',
  components: {
    TipeIcon
  },
  props: {
    text: vueTypes.string
  },
  data: function() {
    return {
      open: false
    }
  },
  methods: {
    onClick() {
      this.open = !this.open
    }
  }
}
</script>

<style scoped lang="postcss">
.content {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
  width: 199px;

  & .row {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 2.125rem;
    width: 100%;
    border-bottom: 1px solid #eeeeee;
    margin-bottom: 4px;

    & .icon {
      display: flex;
      flex: 0 0 2rem;
      transition: transform 250ms ease-in-out;

      &.open {
        transform: rotate(-180deg);
      }
    }
    & p {
      margin: 0;
      font-size: 0.6875rem;
      color: #dce0e5;
    }
  }
}
</style>
