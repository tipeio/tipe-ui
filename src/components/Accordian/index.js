export { default } from './Accordian.vue'
