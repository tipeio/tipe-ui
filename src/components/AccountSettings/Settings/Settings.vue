<template>
  <accordian text="Settings"><tipe-switch 
    text="File Sharing" 
    sublabel="Turn on to share your file changes and updates" /><tipe-switch text="Backup" /></accordian>
</template>

<script>
import Accordian from '../../Accordian'
import TipeSwitch from '../../Switch'

export default {
  name: 'TipeSettings',
  components: {
    Accordian,
    TipeSwitch
  },
  data: function() {
    return {}
  }
}
</script>
