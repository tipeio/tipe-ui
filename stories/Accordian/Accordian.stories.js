import { storiesOf } from '@storybook/vue'

import Accordian from '../../src/components/Accordian'
import { ActivityItem } from '../../src/components/AccountSettings/RecentActivity'

const item = () => ({ title: 'Sam Smith update API key', time: '1 hour ago' })
const style = () => ({ display: 'grid', 'grid-row-gap': '1rem' })

storiesOf('Accordian', module)
  .add('accordian', () => ({
    components: { Accordian, ActivityItem },
    computed: { item },
    template:
      '<accordian text="Recent Activity"><activity-item :item="item"/></accordian>'
  }))
  .add('two accordians', () => ({
    components: { Accordian, ActivityItem },
    computed: { item, style },
    template:
      '<div :style="style"><accordian text="Recent Activity"><activity-item :item="item"/></accordian><accordian text="Recent Activity"><activity-item :item="item"/></accordian></div>'
  }))
