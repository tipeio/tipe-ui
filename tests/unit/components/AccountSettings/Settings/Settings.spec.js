import { shallowMount } from '@vue/test-utils'
import Settings from '@/components/AccountSettings/Settings'
import Accordian from '@/components/Accordian'

describe('Settings.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Settings)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the accordian component', () => {
    const wrapper = shallowMount(Settings)
    expect(wrapper.contains(Accordian)).toBe(true)
  })
})
