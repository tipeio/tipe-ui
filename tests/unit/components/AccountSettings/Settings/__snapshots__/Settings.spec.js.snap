// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings.vue renders 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
  "version": 2.5,
}
`;
